src/bin/cargo/commands/describe_future_incompatibilities.rs

```diff
@@ -0,0 +1,68 @@
+use crate::command_prelude::*;
+use anyhow::anyhow;
+use cargo::core::compiler::future_incompat::{OnDiskReport, FUTURE_INCOMPAT_FILE};
+use cargo::drop_eprint;
+use std::io::Read;
+use uuid::Uuid;
+
+pub fn cli() -> App {
+    subcommand("describe-future-incompatibilities")
+        .arg(
+            opt(
+                "id",
+                "identifier of the report [generated by a Cargo command invocation",
+            )
+            .value_name("id")
+            .required(true),
+        )
+        .about("Reports any crates which will eventually stop compiling")
+}
+
+pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
+    if !config.cli_unstable().future_incompat_report {
+        return Err(anyhow!(
+            "`cargo describe-future-incompatibilities` can only be used on the nightly channel \
+            with `-Z future-incompat-report`"
+        )
+        .into());
+    }
+
+    let ws = args.workspace(config)?;
+    let report_file = ws.target_dir().open_ro(
+        FUTURE_INCOMPAT_FILE,
+        ws.config(),
+        "Future incompatible report",
+    )?;
+
+    let mut file_contents = String::new();
+    report_file
+        .file()
+        .read_to_string(&mut file_contents)
+        .map_err(|e| anyhow!("Failed to read report: {:?}", e))?;
+    let on_disk_report: OnDiskReport<'_> = serde_json::from_str(&file_contents).unwrap();
+
+    let mut warning_message = None;
+    let raw_id = args.value_of("id").unwrap();
+    let bad_id = match Uuid::parse_str(raw_id) {
+        Ok(id) => on_disk_report.uuid != id,
+        Err(_) => true,
+    };
+
+    if bad_id {
+        warning_message = Some(format!("Expected an id of `{}`, but `{}` was provided on the command line. Your report may have been overwritten by a different one.", on_disk_report.uuid, raw_id));
```

Comment:
In my experience, it can be extremely frustrating when a command refuses to display information that I know it has access to. I think it's better to err on the side of showing more information rather than less.

Reply:
The purpose of the `--id` was to handle a case where you get one set of warnings but then in the interim a separate set of warnings is generated. The whole purpose of `--id` is to get the guaranteed same set of warnings from a previous build. I understand your frustration, but in my mind getting *misleading* informatiton trumps that concern and means we should not display anything here if the wrong `--id` is passed.

Reply:
I've changed the code to bail out if a bad 'id' is specified.